src/supervisedoutputconnector.h

```diff
@@ -161,10 +163,11 @@ namespace dd
     void init(const APIData &ad)
     {
       APIData ad_out = ad.getobj("parameters").getobj("output");
-      if (ad_out.has("best"))
-        _best = ad_out.get("best").get<int>();
+      auto output_params
+          = ad_out.createSharedDTO<PredictOutputParametersDto>();
+      _best = output_params->best;
       if (_best == -1)
-        _best = ad_out.get("nclasses").get<int>();
+        _best = output_params->nclasses;
```

Comment:
Is there a case where `nclasses` is defined by the user in the `parameters > output` block?

Reply:
Not defined by the user, but added from https://github.com/jolibrain/deepdetect/blob/master/src/backends/caffe/caffelib.cc#L3656

Reply:
Ok but we are in the `init()` method of the supervised connector here, and I don't think the mllib ever has the opportunity to modify this parameter. So unless `SupervisedOutputConnector::init()` is called from somewhere else at prediction time, nclasses may always be default / not initialized.